include/proto.h

```diff
@@ -491,6 +492,16 @@ enum {
    SL_FLICKER_OFF      = 2,         // Turn flickering off
 };
 
+/* Special item types */
+enum special_type {
+    SPECIAL_TYPE_NONE      = 0,
+    SPECIAL_TYPE_MAGIC     = 1,
+    SPECIAL_TYPE_RARE      = 2,
+    SPECIAL_TYPE_LEGENDARY = 3,
+    SPECIAL_TYPE_UNKNOWN   = 4,
+    SPECIAL_TYPE_CURSED    = 5,
+};
```

Comment:
Enums are unique values, meaning that for example, you could never have magic and cursed at the same time.  Just making sure that's what you want, because it can't easily be changed.  If they could be combined, you'd make these bitflags instead.

Reply:
I thought about this and I think it is ok if they are discrete values.  The main issue this PR tries to address is to quickly identify desirable items without reading the item's description.  Unknown (unrevealed) should always come first, then cursed, then a single value of magic, rare, or legendary as applicable.

I could see some small edge cases where perhaps there's a legendary weapon that is cursed which would be interesting to know about.  But I think that's an outlier case and isn't really what this PR's intent is about.

The text colors should help answer the question: "Should I pick up this item?"